spec/classes/hyperglass_agent_spec.rb

```diff
@@ -2,17 +2,131 @@
 
 describe 'hyperglass::agent' do
   let :node do
-    'rspec.puppet.com'
+    'rspec.example.com'
   end
 
   on_supported_os.each do |os, facts|
     context "on #{os} " do
       let :facts do
-        facts
+        facts.merge(
+          python3_version: '3.6.8'
+        )
       end
 
       context 'with all defaults' do
         it { is_expected.to compile.with_all_deps }
+
+        it {
+          is_expected.to contain_file('/opt/hyperglass').with(
+            {
+              'ensure' => 'directory',
+              'owner'  => 'root',
+              'group'  => 'root',
+              'mode'   => '0755',
+            }
+          )
```

Comment:
I generally like chaining more. It's compacter and also works nicely if you add other lines like `.that_requires()`
```suggestion
          is_expected.to contain_file('/opt/hyperglass').
            with_ensure('directory').
            with_owner('root').
            with_group('root').
            with_mode('0755')
```

Reply:
I disagree, mostly because I can copy/paste from the actual code and write the tests faster. I did leave the .with_this and .with_that for lines that were not too long.